src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,227 @@
+        else
+            this.underlined.set(underlined);
+        return this;
+    }
+
+    public boolean isStrikethrough() {
+        return strikethrough != null && strikethrough.get();
+    }
+
+    public ChatComponent setStrikethrough(boolean strikethrough) {
+        if (this.strikethrough == null)
+            this.strikethrough = new AtomicBoolean(strikethrough);
+        else
+            this.strikethrough.set(strikethrough);
+        return this;
+    }
+
+    public boolean isObfuscated() {
+        return obfuscated != null && obfuscated.get();
+    }
+
+    public ChatComponent setObfuscated(boolean obfuscated) {
+        if (this.obfuscated == null)
+            this.obfuscated = new AtomicBoolean(obfuscated);
+        else
+            this.obfuscated.set(obfuscated);
+        return this;
+    }
+
+    public JsonObject asJson() {
+        JsonObject json = new JsonObject();
+        if (text != null) {
+            json.addProperty("text", text);
+        }
+        if (translate != null) {
+            json.addProperty("translate", translate);
+            JsonArray array = new JsonArray();
+            with.forEach(array::add);
+            json.add("with", array);
+        }
+        if (scoreUsername != null && scoreObjective != null) {
+            JsonObject score = new JsonObject();
+            score.addProperty("name", scoreUsername);
+            score.addProperty("objective", scoreObjective);
+            json.add("score", score);
+        }
+        if (selector != null) {
+            json.addProperty("selector", selector);
+        }
+        if (!extra.isEmpty()) {
+            JsonArray extra = new JsonArray();
+            this.extra.forEach(e -> extra.add(e.asJson()));
+            json.add("extra", extra);
+        }
+        if (bold != null) {
+            json.addProperty("bold", bold.get());
+        }
+        if (italic != null) {
+            json.addProperty("italic", italic.get());
+        }
+        if (underlined != null) {
+            json.addProperty("underlined", underlined.get());
+        }
+        if (strikethrough != null) {
+            json.addProperty("strikethrough", strikethrough.get());
+        }
+        if (obfuscated != null) {
+            json.addProperty("obfuscated", obfuscated.get());
+        }
+        if (color != null) {
+            json.addProperty("color", color.name().toLowerCase());
+        }
+        if (clickEvent != null) {
+            json.add("clickEvent", this.clickEvent.asJson());
+        }
+        if (hoverEvent != null) {
+            json.add("hoverEvent", this.hoverEvent.asJson());
+        }
+        if (insertion != null) {
+            json.addProperty("insertion", insertion);
+        }
+        return json;
+    }
+
+    public String toString() {
+        return asJson().toString();
+    }
+
+    public static ChatComponent create() {
+        return new ChatComponent();
+    }
+
+    public static ChatComponent empty() {
```

Comment:
Ah, you are right. Then the code that uses a purely empty chat string will need to create its own empty chat constant in TridentSDK/Trident#53